Src/LogData.c

```diff
@@ -36,316 +73,209 @@ void writeToEEPROM(uint8_t* buffer, uint16_t bufferSize, uint16_t memAddress, ui
+    //TODO: make sure memAddress is correct, increment address
 }
 
-void readAllDataFromEEPROM(uint16_t addr, AllData* rtn, uint16_t memAddress, uint16_t timeout)
+/**
+ * @brief reads logEntry-sized char array from EEPROM and updates theLogEntry
+ * @param memAddress, the address at which the EEPROM is instructed to start reading
+ */
+void readLogEntryToEEPROM(uint16_t memAddress)
 {
-    char dataRead[200];
+    char dataRead[logEntrySize];
   
-    checkEEPROMBlocking(uint16_t timeout)
-    readFromEEPROM(dataRead, 80, uint16_t memAddress, uint16_t timeout);
-    // need: memAdress and timeout
-
-    rtn->accelGyroMagnetismData_ ->accelX_ = dataRead[0];
-    rtn->accelGyroMagnetismData_ ->accelY_ = dataRead[4];
-    rtn->accelGyroMagnetismData_ ->accelY_ = dataRead[8];
-    rtn->accelGyroMagnetismData_ ->gyroX_ = dataRead[12];
-    rtn->accelGyroMagnetismData_ ->gyroY_ = dataRead[16];
-    rtn->accelGyroMagnetismData_ ->gyroZ_ = dataRead[20];
-    rtn->accelGyroMagnetismData_ ->magnetoX_ = dataRead[24];
-    rtn->accelGyroMagnetismData_ ->magnetoY_ = dataRead[28];
-    rtn->accelGyroMagnetismData_ ->magnetoZ_ = dataRead[32];
-
-    rtn->barometerData_ ->pressure_ = dataRead[36];
-    rtn->barometerData_ ->temperature_ = dataRead[40];
-
-    rtn->combustionChamberPressureData_ ->pressure_ = dataRead[44];
-
-    rtn->gpsData_ ->time_ = dataRead[48];
-    rtn->gpsData_->latitude_.degrees_ = dataRead[52];
-    rtn->gpsData_->latitude_.minutes_ = dataRead[56];
-    rtn->gpsData_->longitude_.degrees_ = dataRead[60];
-    rtn->gpsData_->longitude_.minutes_ = dataRead[64];
-
-    rtn->gpsData_->antennaAltitude_.altitude_ = dataRead[68];
-    rtn->gpsData_->geoidAltitude_.altitude_ = dataRead[72];
-    rtn->gpsData_->totalAltitude_.altitude_ = dataRead[76];
-
-    rtn->oxidizerTankPressureData_ ->pressure_ = dataRead[80];
-
+    checkEEPROMBlocking();
+    readFromEEPROM(dataRead, sizeof(dataRead), memAddress);
+    //TODO: memAdress may not be corrrect bc it's never incremented
+
+    theLogEntry.accelX = dataRead[0];
+    theLogEntry.accelY = dataRead[4];
+    theLogEntry.accelY = dataRead[8];
+    theLogEntry.gyroX = dataRead[12];
+    theLogEntry.gyroY = dataRead[16];
+    theLogEntry.gyroZ = dataRead[20];
+    theLogEntry.magnetoX = dataRead[24];
+    theLogEntry.magnetoY = dataRead[28];
+    theLogEntry.magnetoZ = dataRead[32];
+    theLogEntry.barometerPressure = dataRead[36];
+    theLogEntry.barometerTemperature = dataRead[40];
+    theLogEntry.combustionChamberPressure = dataRead[44];
+    theLogEntry.oxidizerTankPressure = dataRead[48];
+    theLogEntry.gps_time = dataRead[52];
+    theLogEntry.latitude_degrees = dataRead[56];
+    theLogEntry.latitude_minutes = dataRead[60];
+    theLogEntry.longitude_degrees = dataRead[64];
+    theLogEntry.longitude_minutes = dataRead[68];
+    theLogEntry.altitude = dataRead[72];
+    theLogEntry.currentFlightPhase = dataRead][76];
+    theLogEntry.tick = dataRead[80];
```

Comment:
Why are we storing the data in `theLogEntry`?
We should probably be using an "output" argument here, which is simply a pointer to data that the caller allocates, which this function would "output" to. Example:
```c
/**
 * @brief Initializes a LogEntry variable to all 1's.
 * @param[in] output    Pointer to the LogEntry struct to initialize.
 */
void initLogEntry(LogEntry* output) {
    // Should replace this loop with a memcpy(...)
    for (int i = 0; i < sizeof(LogEntry); i++) {
        // Set every byte to all 1's
        ((char*)output)[i] = 0xFF; // = 255 = 0b11111111
    }
}

// Use case
int main() {
    ...
    LogEntry log;
    // Log currently uninitialized (maybe all 0's, maybe random garbage data)

    initLogEntry(&log);
    // Log entry now all 1's (each byte is 0xFF = 0b11111111)
    ...
}
```
This way the calling function has control over the allocation of the data. If we use these kinds of arguments, then we can probably remove the global variable in its entirety.

If you're wondering why we don't just declare the variable locally within the function and return it, then think about the following: 
- What would happen if the argument wasn't a struct, but rather an array (which is really just a pointer).
- What type of memory the array data is stored in if we declare it locally.
- What actually gets returned.

This was a recruiting question I got asked at the career fair in 2019/2020 by Synopsis. Definitely give it a good few minutes of thought before googling it, or ask me and I'll explain it :) (it's actually pretty interesting)